dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/datavalue/DataValueServiceIntegrityTest.java

```diff
@@ -54,88 +43,39 @@
  */
 class DataValueServiceIntegrityTest extends SingleSetupIntegrationTestBase
 {
-    @Autowired
-    private CategoryService categoryService;
-
     @Autowired
     private DataValueService dataValueService;
 
     @Autowired
-    private OrganisationUnitService organisationUnitService;
-
-    @Autowired
-    private DataElementService dataElementService;
-
-    private DataElement deA;
-
-    private Period peA;
-
-    private OrganisationUnit ouA;
-
-    private CategoryCombo categoryComboAB;
-
-    private CategoryCombo categoryComboAC;
-
-    private Category categoryA;
-
-    private Category categoryB;
+    private MetadataSetupServiceExecutor setupExecutor;
 
-    private CategoryOptionCombo categoryOptionComboAB;
-
-    private CategoryOptionCombo categoryOptionComboAC;
+    private final MetadataSetup setup = new MetadataSetup();
```

Comment:
Should perhaps the `MetadataSetup` be freshly instantiated for each test to avoid pollution in between tests?

Reply:
This should not be needed here. The object is only changed by the `setUpTest` method. Tests are just reading its state.